### ISSUE TITLE
Bug in the plugin

### ISSUE THREAD

**Issue by isovector** [isovector/polysemy#30]
The following program crashes the plugin:

```haskell
{-# LANGUAGE TemplateHaskell #-}

module Polysemy.KVStore where

import Polysemy

data KVStore k v m a where
  LookupKV :: k -> KVStore k v m (Maybe v)
  UpdateKV :: k -> Maybe v -> KVStore k v m ()

makeSem ''KVStore

foo :: Member (KVStore k v) r => Sem r (Maybe Bool)
foo = do
  lookupKV "hello"
```

instead, it should be rejected saying couldnt prove `k ~ String` or `v  ~ Bool`

**Comment by isovector** [comment]
This won't be too hard, just need to keep track of which constraint is wanted and which is given. The wanted must be more polymorphic than the given in order to be substituted.

**Comment by isovector** [comment]
Closed in `c3e13cadc70b41344b72be3c940209d7257c5330`
